frontend/class-frontend.php

```diff
@@ -1147,6 +1147,8 @@ public function metakeywords() {
 			$keywords = WPSEO_Meta::get_value( 'metakeywords' );
 			if ( $keywords === '' && ( is_object( $post ) && ( ( isset( $this->options[ 'metakey-' . $post->post_type ] ) && $this->options[ 'metakey-' . $post->post_type ] !== '' ) ) ) ) {
 				$keywords = wpseo_replace_vars( $this->options[ 'metakey-' . $post->post_type ], $post );
+			} else {
+				$keywords = wpseo_replace_vars( $keywords, $post );
 			}
```

Comment:
Hi @jrfnl, in my case a function call (to `wpseo_replace_vars`) will only be performed when `$keywords` and `$post` are not empty. For performance this fallback feels better for me. 

In your case the function call will be executed twice when both values are empty. For performance reasons I would say my solution is better. Please correct me I'm wrong.